src/pycrunch/__init__.py

```diff
@@ -243,15 +268,17 @@ def connect_with_token(token, site_url="https://us.crunch.io/api/",
 
     Returns the API Root Entity, or errors if unable to connect.
     """
-    global session
-    ret = session_class(
-        token=token,
-        domain=urllib.parse.urlparse(site_url).netloc,
-        progress_tracking=progress_tracking
-    ).get(site_url).payload
-    if session is None:
-        session = ret
-    return ret
+    warnings.warn(
+        "connect_with_token will be removed soon. Please use connect instead",
+        UserWarning
+    )
+
+    return connect(
```

Comment:
Is this equivalent to what we had before?

Reply:
pretty much. both functions just called `session_class` if that is what you are asking